### PR TITLE
docs: Fix dead link in documentation

### PR DESCRIPTION
**Pull request by remicres**
### Motivation

Fixed a dead link in the documentation.

### Modifications

Changed the link (the expr-lang repository has moved).

### CODE REVIEW

File: docs/variables.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ The tag is substituted with the result of evaluating the tag as an expression.
 Note that any hyphenated parameter names or step names will cause a parsing error. You can reference them by
 indexing into the parameter or step map, e.g. `inputs.parameters['my-param']` or `steps['my-step'].outputs.result`.
 
-[Learn about the expression syntax](https://github.com/antonmedv/expr/blob/master/docs/Language-Definition.md).
+[Learn about the expression syntax](https://github.com/expr-lang/expr/blob/master/docs/language-definition.md).
```
Comment (terrytangyuan) [inline docs/variables.md:59]:
```suggestion
[Learn about the expression syntax](https://github.com/antonmedv/expr/blob/master/docs/language-definition.md).
```

Comment (terrytangyuan) [reply docs/variables.md:59]: We prefer the original repo

**Review by terrytangyuan — APPROVED** [review]

File: docs/variables.md (hunk 1 above)
Comment (agilgur5) [reply docs/variables.md]: It was actually officially moved to the `expr-lang` org recently: https://github.com/expr-lang/expr/issues/481

We should also probably link to the rendered docs too, which are nicer to look at: https://expr-lang.org/docs/language-definition

Comment (agilgur5) [reply docs/variables.md]: Changed in #12617

**Pull request merged**